src/test/java/com/snowflake/kafka/connector/internal/SinkServiceIT.java

```diff
@@ -170,6 +172,68 @@ public void testIngestion() throws Exception {
     //    assert !conn.pipeExist(pipe);
   }
 
+  @Test
+  public void testTombstoneRecords_ingestion() throws Exception {
+    conn.createTable(table);
+    conn.createStage(stage);
+    SnowflakeSinkService service =
+        SnowflakeSinkServiceFactory.builder(conn)
+            .setRecordNumber(1)
+            .addTask(table, topic, partition)
+            .build();
+
+    SnowflakeConverter converter = new SnowflakeJsonConverter();
+    SchemaAndValue input = converter.toConnectData(topic, null);
+    long offset = 0;
+
+    SinkRecord record1 =
+        new SinkRecord(
+            topic, partition, Schema.STRING_SCHEMA, "test", input.schema(), input.value(), offset);
+
+    service.insert(record1);
+    TestUtils.assertWithRetry(
+        () ->
+            conn.listStage(
+                        stage,
+                        FileNameUtils.filePrefix(TestUtils.TEST_CONNECTOR_NAME, table, partition))
+                    .size()
+                == 1,
+        5,
+        4);
+    service.callAllGetOffset();
+    List<String> files =
+        conn.listStage(
+            stage, FileNameUtils.filePrefix(TestUtils.TEST_CONNECTOR_NAME, table, partition));
+    String fileName = files.get(0);
+
+    assert FileNameUtils.fileNameToTimeIngested(fileName) < System.currentTimeMillis();
+    assert FileNameUtils.fileNameToPartition(fileName) == partition;
+    assert FileNameUtils.fileNameToStartOffset(fileName) == offset;
+    assert FileNameUtils.fileNameToEndOffset(fileName) == offset;
+
+    // wait for ingest
+    TestUtils.assertWithRetry(() -> TestUtils.tableSize(table) == 1, 30, 20);
+
+    ResultSet resultSet = TestUtils.showTable(table);
+    LinkedList<String> contentResult = new LinkedList<>();
+    while (resultSet.next()) {
+      contentResult.add(resultSet.getString("RECORD_CONTENT"));
+    }
+    resultSet.close();
+
+    assert contentResult.size() == 1;
+
+    ObjectNode emptyNode = MAPPER.createObjectNode();
+    assert contentResult.get(0).equalsIgnoreCase(emptyNode.toString());
```

Comment:
assert that there is only one result too maybe?

Reply:
line 224